### PR TITLE
User can edit an entry - Tiffany Brown

### PR DESCRIPTION
**Pull request by tiffanydbrown**
Message in Slack.

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -23,38 +29,65 @@ function handleSubmit(event) {
...
 
-  formValues.entryId = data.nextEntryId;
-  data.nextEntryId++;
-  data.entries.unshift(formValues);
-
-  $photo.setAttribute('src', 'images/placeholder-image-square.jpg');
-
-  $form.reset();
-
   $ul.prepend(renderEntry('entries'));
```
Comment (shawnkost) [inline js/main.js]: Delete this. We only want to create an entry if `data.editing === null`

File: js/main.js [hunk 2]
```diff
@@ -23,38 +29,65 @@ function handleSubmit(event) {
...
   viewSwap($entryView);
 
   toggleNoEntries();
+
+  if (data.editing === null) {
+    renderEntry('entries');
```
Comment (shawnkost) [inline js/main.js]: You need to append the result of `renderEntry` to your `ul`. Right now, you're creating an entry, but aren't actually adding it to the webpage

File: js/main.js [hunk 3]
```diff
@@ -23,38 +29,65 @@ function handleSubmit(event) {
...
+    formValues.entryId = data.nextEntryId;
+    data.nextEntryId++;
+    data.entries.unshift(formValues);
+    console.log(formValues.entryId);
```
Comment (shawnkost) [inline js/main.js]: Remove all `console.log`s

File: js/main.js [hunk 4]
```diff
@@ -23,38 +29,65 @@ function handleSubmit(event) {
...
+    for (let i = 0; i < data.entries.length; i++) {
+      if (data.entries[i].entryId === formValues.entryId) {
+        data.entries[i] = formValues;
+      }
+    }
+
+    const newDOM = renderEntry(formValues);
+    const existingLi = document.querySelector(
+      `li[data-entry-id=" data.editing"]`
+    );
```
Comment (shawnkost) [inline js/main.js]: This isn't going to work. You need to interpolate your `data.editing.entryId` into the template literal

File: js/main.js [hunk 5]
```diff
@@ -23,38 +29,65 @@ function handleSubmit(event) {
...
+
+    if (existingLi) {
+      existingLi.replaceWith(newDOM);
+      const $updateEntry = document.getElementById('entry-form');
+      $updateEntry.innerText = 'New Entry';
```
Comment (shawnkost) [inline js/main.js]: I don't think this is the correct element? Your `div` with the `data-view="entry-form"` is getting selected by this. You want your `h2` that says "New entry" to be the one that gets targeted by this

**Review by shawnkost — CHANGES_REQUESTED** [review]
Some changes here Tiffany!

**Comment by tiffanydbrown** [comment]
https://github.com/tiffanydbrown/code-journal/assets/139157669/7ee4c9d7-7d40-4fdc-a691-58ff5c42ef32

File: index.html [hunk 6]
```diff
@@ -14,6 +14,9 @@
     <link rel="stylesheet" href="css/reset.css" />
     <link rel="stylesheet" href="css/layout.css" />
     <link rel="stylesheet" href="css/styles.css" />
+    <script
+      src="https://kit.fontawesome.com/3d1b6d9784.js"
+      crossorigin="anonymous"></script>
```
Comment (shawnkost) [inline index.html]: Not needed. This was already included for you

File: js/main.js [hunk 7]
```diff
@@ -23,38 +29,63 @@ function handleSubmit(event) {
...
+      if (data.entries[i].entryId === formValues.entryId) {
+        data.entries[i] = formValues;
+      }
+    }
+
+    const newDOM = renderEntry(formValues);
+    const existingLi = document.querySelector(
+      `li[data-entry-id="${data.editing.entryId}"]`
+    );
+    console.log(existingLi);
```
Comment (shawnkost) [inline js/main.js]: Remove all `console.log`s!

File: js/main.js [hunk 8]
```diff
@@ -63,11 +94,37 @@ function renderEntry(entry) {
...
+      if (data.entries[i].entryId === Number($clickedEntryId)) {
+        data.editing = data.entries[i];
+        showFormAndEditEntry(data.entries[i]);
+      }
+    }
+  }
+});
+
+function showFormAndEditEntry(entry) {
+  console.log(entry);
```
Comment (shawnkost) [inline js/main.js]: Remove all `console.log`s!

**Review by shawnkost — CHANGES_REQUESTED** [review]
Few changes here Tiffany!

**Review by shawnkost — APPROVED** [review]
👍🏻

**Pull request merged**